operator/java-client/src/main/java/org/infinispan/App.java

```diff
@@ -0,0 +1,75 @@
+package org.infinispan;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.configuration.ConfigurationBuilder;
+import org.infinispan.client.hotrod.impl.ConfigurationProperties;
+
+import gnu.getopt.Getopt;
+
+/**
+ * Infinispan+operator+hotrod client quickstart
+ *
+ */
+public class App {
+    static String host = "127.0.0.1";
+    static int port = ConfigurationProperties.DEFAULT_HOTROD_PORT;
+    static String username, password;
+
+    public static void main(String[] args) {
+        // getopt and configure static variables
+        configure(args);
+        // Create a configuration for a locally-running server
+        ConfigurationBuilder builder = new ConfigurationBuilder();
+        builder.addServer().host(host).port(port);
+        if (username != null && password != null) {
+            builder.security().authentication().saslMechanism("PLAIN").username(username).password(password);
+        }
+        // Connect to the server
+        RemoteCacheManager cacheManager = new RemoteCacheManager(builder.build());
+        // Obtain the remote cache
+        try {
+        cacheManager.administration().createCache("quickstart-cache", "org.infinispan.DIST_SYNC");
+        } catch (Exception ex) {
+            // Maybe cache already exist? Go on anyway
+        }
+        RemoteCache<String, String> cache = cacheManager.getCache("quickstart-cache");
+        /// Store a value
+        cache.put("key", "value");
+        // Retrieve the value and print it out
+        System.out.printf("key = %s\n", cache.get("key"));
+        // Stop the cache manager and release all resources
+        cacheManager.administration().removeCache("quickstart-cache");
+        cacheManager.stop();
+    }
+
+    static void configure(String args[]) {
+        Getopt g = new Getopt("testprog", args, "h:p:U:P:");
```

Comment:
The code would be simpler if you just used system properties for the different parameters you want to pass in. It'd also remove a dependency. The focus here is not so much the validation part, but showing that you can just use plain Java client code to talk to Infinispan started by operator.

Also, given that authentication is enabled by default, I'd check that user and password system properties are given by the user.

Eventually, the development profile would avoid the need for username and password for the simplest of example, but you can keep a very simple separate example just for that.

Reply:
System vars sound like a good idea, will do.
I need this branch stable atm, will do change later.